lib/sim.c

```diff
@@ -1181,7 +1247,102 @@ sim_simulate(sim_t *self, uint64_t max_events)
     return ret;
 }
 
+int
+sim_simulate_arg(sim_t *self) 
+{
+    int ret = 0;
+    double recomb_rate, coal_rate, total_rate; 
+    individual_t **S = self->intersected_buffer;
+    unsigned int S_size = 0;
+    avl_node_t *node;
+    uintptr_t int_ptr;
+    uint64_t id;
+    individual_t *ind;
+    /* An arbitrary radius and parental location for using existing spatial 
+     * functions to complete events */
+    double z[] = {self->pixel_size / 2.0, self->pixel_size / 2.0}, r = self->pixel_size / 4.0;
+    if(self->dimension == 1) {
+        z[1] = 0.0;
+    }
+    while(self->ancestral_material > self->num_loci) {
+        coal_rate = 2.0 * self->Ne * gsl_sf_choose(((set_map_value_t *)self->Q.head->item)->key, 2);
+        recomb_rate = 0.0;
+        S_size = 0;
+        for(node = self->P[0].head; node != NULL; node = node->next) {
+            id = *((uint64_t *) node->item);
+            int_ptr = (uintptr_t) id;
+            ind = (individual_t *) int_ptr;
+            recomb_rate += (double)(((set_map_value_t *)ind->ancestry.tail->item)->key 
+                - ((set_map_value_t *)ind->ancestry.head->item)->key) * self->rho;
+            S[S_size] = ind;
+            S_size++;
+        }
+        total_rate = coal_rate + recomb_rate;
+        self->time += gsl_ran_exponential(self->rng, 1.0 / total_rate);
+        if(gsl_rng_uniform(self->rng) < coal_rate / total_rate) {
+            // Coalescence event
+            self->num_parents = 1;
+            self->num_children = 2;
+            gsl_ran_choose(self->rng, self->child_buffer, self->num_children, S, S_size, sizeof(individual_t *));
+            ret = sim_complete_event(self, z, r, 0, 0);
+            if (ret != 0) {
+                goto out;
+            }
+        }
+        else {
+            // Recombination event
+            self->num_parents = 2;
+            self->num_children = 1;
+            gsl_ran_choose(self->rng, self->child_buffer, self->num_children, S,
+                        S_size, sizeof(individual_t *));
+            ret = sim_complete_event(self, z, r, 1, 0);
+            if (ret != 0) {
+                goto out;
+            }
+        }
+    }
+out:
+    return ret;
+}
 
+/*
+ * Moves the whole population into the first pixel and overwrites
+ * spatial parameters so that it is impossible to intersect any 
+ * other pixel in order to connect the spatial and panmictic
+ * simulations. */
+int
+sim_setup_arg(sim_t *self)
+{
+    double ret = 0;
+    avl_node_t *node;
+    uintptr_t int_ptr;
+    uint64_t id;
+    individual_t *ind;
+    unsigned int pixel_occ, i, j;
+    self->num_parents = 1;
+    self->num_children = 1;
+    double z[] = {self->pixel_size / 2.0, self->pixel_size / 2.0}, r = self->pixel_size / 4.0;
```

Comment:
Yes, that's true... I guess I'm just nervous as these are treated as parameters in the spatial simulation and modifying parameters makes me nervous. You're right though, this is probably the simplest way to achieve this and reusing the existing infrastructure as much as possible.